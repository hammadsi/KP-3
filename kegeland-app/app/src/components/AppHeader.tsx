import {StackHeaderProps} from '@react-navigation/stack';
import React from 'react';
import {StyleSheet} from 'react-native';
import Icon from 'react-native-vector-icons/MaterialCommunityIcons';
import {TouchableOpacity} from 'react-native-gesture-handler';
import {Appbar, Text, useTheme} from 'react-native-paper';

export type AppHeaderProps = StackHeaderProps;

/**
 * Component for rendering the applications main header
 * @param props the props
 * @see {@link AppHeaderProps}
 * @see {@link Appbar.Header}
 */
const AppHeader: React.FC<AppHeaderProps> = ({navigation, back}) => {
  const {colors} = useTheme();
  return back ? (
    <Appbar.Header
      style={(styles.wrapper, {backgroundColor: colors.background})}>
      <TouchableOpacity style={styles.back} onPress={() => navigation.goBack()}>
        <Icon
          size={30}
          style={styles.backIcon}
          color="#0081cb"
          name="chevron-left"
        />
        <Text style={styles.backTitle}>{back.title}</Text>
      </TouchableOpacity>
    </Appbar.Header>
  ) : null;
};

const styles = StyleSheet.create({
  wrapper: {
    padding: 0,
    justifyContent: 'flex-start',
  },
  back: {
    flex: 1,
    flexDirection: 'row',
    alignSelf: 'center',
    alignItems: 'center',
    alignContent: 'center',
  },
  backIcon: {
    paddingTop: 3,
  },
  backTitle: {
    color: '#0081cb',
    fontWeight: '500',
    fontSize: 18,
  },
});

export default AppHeader;
