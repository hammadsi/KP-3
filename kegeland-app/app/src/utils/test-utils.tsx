/* istanbul ignore file */
import React from 'react';
import {NavigationContainer} from '@react-navigation/native';
import {Provider as PaperProvider} from 'react-native-paper';
import {render, RenderOptions} from '@testing-library/react-native';
import {Provider as ReduxProvider} from 'react-redux';

import {store} from '~state/store';
import {lightTheme} from '~constants/theme';
import {initialStore, mockStore} from '~state/ducks/__mocks__/store.mock';

type TestWrapperProps = {
  children: React.ReactNode;
  store?: typeof store;
};

/**
 * Wrapper component with redux state. Used when testing components
 * which requires redux.
 * @param param0 test wrapper props @see {@link TestWrapperProps}
 * @returns
 */
export const TestWrapper: React.FC<TestWrapperProps> = ({
  children,
  ...props
}) => {
  return (
    <ReduxProvider store={props.store || mockStore(initialStore)}>
      <PaperProvider theme={lightTheme}>
        <NavigationContainer theme={lightTheme}>{children}</NavigationContainer>
      </PaperProvider>
    </ReduxProvider>
  );
};

/**
 * Custom test renderer which wraps components with a redux store provider
 * @param ui the component to render
 * @param options renderer options
 * @returns test rendered componnet
 */
export const customRender = <T extends any>(
  ui: React.ReactElement<T>,
  options?: {
    wrapperProps?: {
      store: any;
    };
    renderOptions?: RenderOptions;
  },
) =>
  render(ui, {
    wrapper: (props) => <TestWrapper {...props} {...options?.wrapperProps} />,
    ...options?.renderOptions,
  });

// re-export everything
export * from '@testing-library/react-native';

// override render method
export {customRender as render};
