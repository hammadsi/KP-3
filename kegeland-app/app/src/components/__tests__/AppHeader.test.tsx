import React from 'react';

import {render} from '~utils/test-utils';

import AppHeader from '../AppHeader';

describe('Test Accordion-component', () => {
  it('should render correctly', () => {
    const props: any = {
      navigation: {
        goBack: jest.fn(),
      },
      back: {title: 'Home'},
    };
    const component = <AppHeader {...props} />;
    const tree = render(component);
    expect(tree.toJSON()).toMatchSnapshot();
  });
  it('should render without back button', () => {
    const props: any = {
      navigation: {
        goBack: jest.fn(),
      },
      back: undefined,
    };
    const component = <AppHeader {...props} />;
    const tree = render(component);
    expect(tree.toJSON()).toMatchSnapshot();
  });
});
