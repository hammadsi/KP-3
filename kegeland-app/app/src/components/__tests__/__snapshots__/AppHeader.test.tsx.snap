// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test Accordion-component should render correctly 1`] = `
<View
  collapsable={false}
  pointerEvents="box-none"
  style={
    Object {
      "flex": 1,
    }
  }
>
  <RCTSafeAreaView
    emulateUnlessSupported={true}
    style={
      Array [
        Object {
          "backgroundColor": "#f6f6f6",
          "elevation": 4,
          "zIndex": 0,
        },
        Object {
          "shadowColor": "#000000",
          "shadowOffset": Object {
            "height": 3,
            "width": 0,
          },
          "shadowOpacity": 0.24,
          "shadowRadius": 4,
        },
      ]
    }
  >
    <View
      collapsable={false}
      nativeID="animatedComponent"
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "#f6f6f6",
          "elevation": 0,
          "flexDirection": "row",
          "height": 56,
          "marginTop": 0,
          "paddingHorizontal": 4,
        }
      }
    >
      <RNGestureHandlerButton
        collapsable={false}
        delayLongPress={600}
        enabled={true}
        exclusive={true}
        handlerTag={1}
        handlerType="NativeViewGestureHandler"
        onGestureEvent={[Function]}
        onGestureHandlerEvent={[Function]}
        onGestureHandlerStateChange={[Function]}
        onHandlerStateChange={[Function]}
        rippleColor={0}
        touchSoundDisabled={false}
      >
        <View
          accessible={true}
          collapsable={false}
          nativeID="animatedComponent"
          style={
            Object {
              "alignContent": "center",
              "alignItems": "center",
              "alignSelf": "center",
              "flex": 1,
              "flexDirection": "row",
              "opacity": 1,
            }
          }
        >
          <Text
            allowFontScaling={false}
            selectable={false}
            style={
              Array [
                Object {
                  "color": "#0081cb",
                  "fontSize": 30,
                },
                Object {
                  "paddingTop": 3,
                },
                Object {
                  "fontFamily": "Material Design Icons",
                  "fontStyle": "normal",
                  "fontWeight": "normal",
                },
                Object {},
              ]
            }
          >
            Û∞ÖÅ
          </Text>
          <Text
            style={
              Array [
                Object {
                  "color": "#000000",
                  "fontFamily": "System",
                  "fontWeight": "400",
                },
                Object {
                  "textAlign": "left",
                },
                Object {
                  "color": "#0081cb",
                  "fontSize": 18,
                  "fontWeight": "500",
                },
              ]
            }
          >
            Home
          </Text>
        </View>
      </RNGestureHandlerButton>
    </View>
  </RCTSafeAreaView>
</View>
`;

exports[`Test Accordion-component should render without back button 1`] = `
<View
  collapsable={false}
  pointerEvents="box-none"
  style={
    Object {
      "flex": 1,
    }
  }
/>
`;
